bionic/code_analyzer.py

```diff
@@ -0,0 +1,134 @@
+"""
+This module implements the logic to analyze your code objects and
+returns the result in the form of a byte array.
+"""
+
+import inspect
+import json
+import warnings
+
+from .utils.misc import oneline
+
+
+DEFAULT_NAME = b"zA)T+ndu/Ygd%JH=eux)[F9:<>)zND]}u4x,c49tWH<yd5+#+4Xg{=J<[S5/'XG+"
+
+
+def analyze_func(func):
+    assert inspect.isroutine(func)
+
+    analyzer = CodeAnalyzer()
+    analyzer.analyze(func)
+    return analyzer.get_analyzed_bytes()
+
+
+class CodeAnalyzer:
```

Comment:
This class seems like a good candidate for tests.

Reply:
Will do!

Reply:
Done!